\documentclass[slidestop,14pt]{beamer}

% TODO:
% Lookup expected QC results (numbers)

\title{Genome of the Netherlands}
\providecommand{\myConference}{Lab-J work discussion}
\providecommand{\myDate}{Wednesday, 8 June 2011}
\author{Martijn Vermaat}
\providecommand{\myGroup}{Leiden Genome Technology Center}
\providecommand{\myDepartment}{Department of Human Genetics}
\providecommand{\myCenter}{Center for Human and Clinical Genetics}
\providecommand{\lastRightLogo}{
  \raisebox{-0.1cm}{
    \includegraphics[scale = 0.055]{lgtc_logo}
  }
}
\providecommand{\lastCenterLogo}{}

\usetheme{lumc}

\begin{document}

% This disables the \pause command, handy in the editing phase.
%\renewcommand{\pause}{}

% Make the title page.
\bodytemplate

%\frame{
%  \frametitle{Genome of the Netherlands}
%  \tableofcontents
%}

\section{Introduction}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\begin{frame}
  \frametitle{Genome of the Netherlands}

  \vspace{\baselineskip}

  Ultra-sharp genetic group portrait of the Dutch

  \pause

  \vspace{\baselineskip}

  Consortium of
  \begin{itemize}
    \item UMCG (Groningen)
    \item Erasmus MC (Rotterdam)
    \item VUMC and AMC (Amsterdam)
    \item UMCU (Utrecht)
    \item LUMC (Leiden)
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Goals}

  \vspace{\baselineskip}

  Medical:
  \begin{itemize}
    \item Enrich biobank data
    \item Construct a Dutch reference genome
    \item Identify rare variants
  \end{itemize}

  \vspace{\baselineskip}

  Scientific:
  \begin{itemize}
    \item Gain high-throughput sequencing experience
    \item Develop new scientific methods
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Setup}

  \vspace{\baselineskip}

  \begin{itemize}
    \item 250 trios of parents and adult child
      % Also 10 twins
    \item Sequencing in China (BGI)
    \item Analysis in the Netherlands
  \end{itemize}

  \vspace{\baselineskip}

  \includegraphics[width=0.3\linewidth,transparent]{parents-girl.png}
  \includegraphics[width=0.3\linewidth,transparent]{parents-boy.png}
  \includegraphics[width=0.3\linewidth,transparent]{parents-girl.png}
\end{frame}

\begin{frame}
  \frametitle{Sequencing method}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Illumina HiSeq
    \item Full-genome
    \item 500M paired reads of 90b
    \item 12x average coverage
  \end{itemize}

  \vspace{0.5\baselineskip}

  \begin{center}
    \includegraphics[width=0.35\linewidth,transparent]{hiseq.png}
  \end{center}
\end{frame}

\section{Sample processing}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\begin{frame}
  \frametitle{Raw sequencing data}

  \vspace{\baselineskip}

  \begin{itemize}
    \item 750 samples
    \item 100GB per sample
    \item 75TB of FASTQ files
  \end{itemize}

  \pause

\begin{center}
{\renewcommand{\arraystretch}{0.6}
\renewcommand{\tabcolsep}{10pt}
\begin{tabular}{p{200pt} r@{.}l}
  & \multicolumn{2}{l}{\footnotesize \# terabytes:}\\
  Human genome & 0&001\\
  Average PC & 0&1\\
  GoNL & \multicolumn{1}{r@{\phantom{.}}}{75}& \pause \\
  3D effects rendering for Avatar & \multicolumn{1}{r@{\phantom{.}}}{1,000}&\\
  Google processes per day & \multicolumn{1}{r@{\phantom{.}}}{24,000}&
\end{tabular}}
\end{center}
\end{frame}

\begin{frame}
  \frametitle{Pipeline}

  \vspace{\baselineskip}

  Developed at UMCG

  \vspace{\baselineskip}

  \begin{itemize}
    \item Alignment
    \item SNP calling
    \item Based on Broad (GATK) pipeline
  \end{itemize}

  \vspace{\baselineskip}

  \url{http://www.bbmriwiki.nl/wiki/SnpCallingPipeline}
\end{frame}

\begin{frame}
  \frametitle{Hardware}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Millipede cluster (RUG)
    \item Also: AMC (grid), UU, EMC
    \item Done in a month from now
  \end{itemize}

  % UMCG: ~250 lanes per week
  % UU and EMC: ~13 lanes per week
  % EMC: 0 lanes per week

  \vspace{0.5\baselineskip}

  \begin{center}
    \includegraphics[width=0.45\linewidth,transparent]{millipedes.jpg}
  \end{center}
\end{frame}

\section{Quality control}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\begin{frame}
  \frametitle{Phases}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Pre-alignment
    \item Alignment
    \item Variant calling
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Pre-alignment}

  \vspace{\baselineskip}

  \begin{itemize}
    %\item Duplicate input checking
    \item FASTQ quality scores
    \item GC\%
    \item N\%
    \item Read length distribution
    \item Over-representation of k-mers
  \end{itemize}

  % http://bioinfo-core.org/index.php/9th_Discussion-28_October_2010
  % http://code.google.com/p/cutadapt/
\end{frame}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{
    \parbox[c][\paperheight]{\paperwidth}{%
      \vfill \begin{center}
      \includegraphics[width=0.9\paperwidth,keepaspectratio=true]{quality.png}
      \end{center} \vfill}
  }
  \frame{}
}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{
    \parbox[c][\paperheight]{\paperwidth}{%
      \vfill \begin{center}
      \includegraphics[width=0.9\paperwidth,keepaspectratio=true]{gc.png}
      \end{center} \vfill}
  }
  \frame{}
}

\begin{frame}
  \frametitle{Alignment}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Percentage aligned
    \item Insert size distribution
    \item Coverage distribution (X/Y/autosomes)
    \item Mapping quality distribution
  \end{itemize}
  % todo: picture of aligned reads?
\end{frame}

\begin{frame}
  \frametitle{Variant calling}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Transition / transversion ratio
    \item Mutation rate (Y/mtDNA/autosomes)
    \item Distribution of SNPs found in dnSNP
    \item Indel / substitution ratio
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Pipeline concordance}

  \vspace{\baselineskip}

  Comparison of pipelines on alignment and variant calling

  \vspace{\baselineskip}

  \begin{itemize}
    \item SOAP pipeline (BGI)
    \item Broad pipeline (UMCG)
    \item GAPSS3 pipeline (LUMC)
    \item (Immuno-chip)
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{SNP call concordance}

  \vspace{\baselineskip}

  Laurent Francioli:
  \begin{itemize}
    \item 50 samples
    \item SOAP (BGI) versus Broad (UMCG)
    \item Site and allele concordance
  \end{itemize}
\end{frame}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{\includegraphics[width=\paperwidth,keepaspectratio=true,clip=true,trim=2cm 10cm 3cm 0]{bgi-concordance.pdf}}
  \frame{}
}

\begin{frame}
  \frametitle{SNP call concordance}

  \vspace{\baselineskip}

  Process 9 samples with GAPSS3 (LUMC)

  \vspace{\baselineskip}

\begin{center}
{\renewcommand{\arraystretch}{1.0}
\renewcommand{\tabcolsep}{10pt}
\begin{tabular}{p{140pt} r}
Pipeline & A4a SNPs\\
\hline
SOAP (BGI) & 3,204,825\\
Broad (UMCG) & 3,769,099\\
GAPSS3 (LUMC) & 3,374,486\\
GAPSS3 filtered & 2,459,515
\end{tabular}}
\end{center}

\end{frame}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{\includegraphics[width=0.65\paperwidth,keepaspectratio=true,clip=true,trim=0 3cm 3cm 5cm]{lumc-comparison.pdf}}
  \frame{}
}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{\includegraphics[width=0.65\paperwidth,keepaspectratio=true,clip=true,trim=0 3cm 3cm 5cm]{lumc-concordance.pdf}}
  \frame{}
}

\begin{frame}
  \frametitle{SNP call concordance}
  \vspace{-\baselineskip}
\begin{center}
{\footnotesize\renewcommand{\arraystretch}{1.2}
\renewcommand{\tabcolsep}{8pt}
\begin{tabular}{r c c c c}
Sites called & BGI SOAP & UMCG & LUMC & LUMC filtered\\
\hline
7,236 & $\bullet$ (0.2\%) & & $\bullet$ (0.2\%) &\\
11,696 & $\bullet$ (0.4\%) & & $\bullet$ (0.3\%) & $\bullet$ (0.5\%)\\
35,678 & & & $\bullet$ (1.1\%) & $\bullet$ (1.5\%)\\
114,468 & & & $\bullet$ (3.4\%) &\\
125,048 & $\bullet$ (3.9\%) & & &\\
143,303 & & $\bullet$ (3.8\%) & $\bullet$ (4.2\%) & $\bullet$ (5.8\%)\\
247,947 & $\bullet$ (7.7\%) & $\bullet$ (6.6\%) & &\\
249,208 & & $\bullet$ (6.6\%) & $\bullet$ (7.4\%) &\\
315,744 & & $\bullet$ (8.4\%) & &\\
544,059 & $\bullet$ (17.0\%) & $\bullet$ (14.4\%) & $\bullet$ (16.1\%) &\\
2,268,838 & $\bullet$ (70.8\%) & $\bullet$ (60.2\%) & $\bullet$ (67.2\%) & $\bullet$ (92.2\%)
\end{tabular}}
\end{center}
\end{frame}

\section{Mitochondrial DNA}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\begin{frame}
  \frametitle{Mitochondria}

  \vspace{\baselineskip}

  50 to hundreds per cell

  \vspace{0.5\baselineskip}

  DNA:
  \begin{itemize}
    \item Circular, 16kbp
    \item 5--10 copies per mt
    \item Maternally inherited
  \end{itemize}

  \vspace{-\baselineskip}

  \hspace{0.6\linewidth}\includegraphics[width=0.3\linewidth,transparent]{mitochondrion.png}
\end{frame}

\begin{frame}
  \frametitle{Study}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Haplogroups
    \item Heteroplasmy
      \begin{itemize}
        \item Sequencing artefacts
        \item De novo
        \item Paternal inheritance?
      \end{itemize}
  \end{itemize}

  % circular alignment
\end{frame}

\begin{frame}
  \frametitle{Early explorations}

  \vspace{\baselineskip}

  \begin{itemize}
    \item High coverage
    \item Allele frequencies
    \item Mapping/read artefacts
  \end{itemize}
\end{frame}

{
  \setbeamercolor{background canvas}{bg=}
  \usebackgroundtemplate{\includegraphics[width=\paperwidth,keepaspectratio=true]{overlaps.png}}
  \frame{}
}

\begin{frame}
  \frametitle{Early explorations}

  \vspace{\baselineskip}

  Position 15171, reference G

  \vspace{\baselineskip}

  A3c (child):
  \begin{itemize}
  %\item Coverage 1979x
    \item 99\% {\color{Red} A}
  \end{itemize}

  \vspace{\baselineskip}

  A3b (mother):
  \begin{itemize}
  %\item Coverage 1863x
    \item 89\% {\color{Red} A}
    \item 10\% {\color{Green} G}
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Early explorations}

%\begin{picture}(0,0)
%\put(100,-80){\includegraphics[width=0.2\paperwidth,keepaspectratio=true]{snp.png}}
%\end{picture}

  \uncover<2>{\includegraphics[width=0.8\paperwidth,keepaspectratio=true]{snp.png}}

  \vspace{-0.214\paperwidth}

  Position 9287, reference A
%82\% A (avg pos 48/90,  0\% unpaired, 39\% reverse, avg readq 30, avg baseq 29, avg mapq 39)  15\% C (avg pos 18/90,  0\% unpaired,  0\% reverse, avg readq 27, avg baseq  6, avg mapq 47)

  \vspace{\baselineskip}

  A4a:
  \begin{itemize}
  %\item Coverage 939x
    \item 82\% {\color{Green} A}
    \item 15\% {\color{Red} C}
    \item Avg position for {\color{Red} C} is 18/90
    \item Low base quality
    \item Allmost all {\color{Red} C} reads on one strand
  \end{itemize}
\end{frame}

\section{Y chromosome}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\begin{frame}
  \frametitle{Analysis}

  \vspace{\baselineskip}

  \begin{itemize}
    \item Focus on 150kbp non-repetitive part
    \item Compare to Y-tree and look for de novo
    \item Focus on Chris Tyler's Y-part
    \item Female reads mapping to Y
  \end{itemize}
\end{frame}

\section{Questions?}

\frame{
  \frametitle{Genome of the Netherlands}
  \tableofcontents[currentsection]
}

\lastpagetemplate
\begin{frame}
  \frametitle{Acknowledgements}

  \vspace{\baselineskip}

  Jeroen Laros\\
  Michiel van Galen\\
  Peter de Knijff\\
  Johan den Dunnen\\
  Gert-Jan van Ommen

  \vspace{\baselineskip}

  The Genome of the Netherlands consortium
\end{frame}

\end{document}
